diff --git a/.gitignore b/.gitignore
index 5614825..bdf3397 100644
--- a/.gitignore
+++ b/.gitignore
@@ -11,3 +11,10 @@ nohup.out
 
 # only for development, normally external
 /doreah
+
+# dev environment and build artifacts
+build
+dist
+*.egg*
+.env
+.python-version
diff --git a/Dockerfile.alp.dev b/Dockerfile.alp.dev
new file mode 100644
index 0000000..a7c0a03
--- /dev/null
+++ b/Dockerfile.alp.dev
@@ -0,0 +1,40 @@
+FROM python:3-alpine as base
+
+WORKDIR /usr/src/app
+
+RUN apk add --no-cache --virtual .build-deps \
+    gcc \
+    libxml2-dev \
+    libxslt-dev \
+    py3-pip \
+    libc-dev \
+    linux-headers
+
+RUN pip3 install psutil
+
+# development
+FROM base as dev
+
+COPY requirements.txt requirements.txt
+RUN pip3 install -r requirements.txt
+
+COPY . .
+# RUN python3 setup.py build
+# RUN python3 setup.py install
+
+RUN apk del .build-deps
+
+EXPOSE 42010
+
+ENTRYPOINT python -m maloja.server
+
+# production
+FROM base
+
+RUN pip3 install --upgrade --no-cache-dir malojaserver
+
+RUN apk del .build-deps
+
+EXPOSE 42010
+
+ENTRYPOINT maloja run
diff --git a/Dockerfile.dev b/Dockerfile.dev
new file mode 100644
index 0000000..974413f
--- /dev/null
+++ b/Dockerfile.dev
@@ -0,0 +1,18 @@
+FROM python:3-buster
+
+WORKDIR /usr/src/app
+
+RUN apt-get update -y
+RUN apt-get install -y python3 python3-pip
+RUN pip3 install psutil
+
+COPY requirements.txt requirements.txt
+RUN pip3 install -r requirements.txt
+
+COPY . .
+RUN python3 setup.py build
+RUN python3 setup.py install
+
+EXPOSE 42010
+
+ENTRYPOINT maloja run
\ No newline at end of file
diff --git a/README.DEV.md b/README.DEV.md
new file mode 100644
index 0000000..4753960
--- /dev/null
+++ b/README.DEV.md
@@ -0,0 +1,26 @@
+# Development using Docker
+
+Start a Docker container for local development.
+
+```bash
+docker build -f Dockerfile.dev -t maloja_dev .
+docker run -it --rm -p 42010:42010 -v /home/$USER/git/maloja:/usr/src/app --env-file .env maloja_dev
+```
+
+Or you can use the equivalent docker-compose command.
+
+```bash
+docker-compose up
+```
+
+## Advantages
+
+- No dependencies installed on host
+
+## TODO:
+
+- Run using maloja.server directly, i.e. `python -m maloja.server` instead of pip install or setup.py build/install
+- Automatically updated content on edits. See these resources:
+  - [Bottle Tutorial Auto Reloading](https://bottlepy.org/docs/dev/tutorial.html#auto-reloading)
+  - [How to change and reload python code in waitress without restarting the server?](https://stackoverflow.com/questions/36817604/how-to-change-and-reload-python-code-in-waitress-without-restarting-the-server)
+- Allow CSV import using dev server
diff --git a/docker-compose.yml b/docker-compose.yml
new file mode 100644
index 0000000..7287f57
--- /dev/null
+++ b/docker-compose.yml
@@ -0,0 +1,13 @@
+version: "3.9"
+services:
+  dev:
+    build:
+      context: .
+      dockerfile: Dockerfile.dev
+    ports:
+      - "42010:42010"
+    volumes:
+      - .:/usr/local/app
+    env_file: .env
+    stdin_open: true
+    tty: true
\ No newline at end of file
